project-ideas/order-ledger/createSalesOrder.md

```diff
@@ -1,47 +1,210 @@
 ### co.hotwax.orderledger.order.OrderServices.create#SalesOrder (Application Layer - OMS)
-This service will take in the order JSON in OMSNewOrdersFeed and set up a complete order by performing any surrounding crud operations as needed.
+
+This is the application layer service which will be responsible for
+1. handling multiple business rules like existence checks, validating customer, facility, products, etc.
+2. coordinating calls to multiple lower level services in the right sequence to achieve the order creation. This includes create Customer, Contact Mechs etc.
+
+This service will take in the order JSON transformed from the response received from an external API request (REST/GraphQL) and create the payload for the core create#org.apache.ofbiz.order.order.OrderHeader service by performing any surrounding validations and crud operations as needed.
+
 1. Parameters
     * Input Parameters
-        * orderJson (Map)
-2. Initialize orderContext
-3. Set following to orderContext
-    * orderContext.externalId = orderJson.externalId
-    * orderContext.orderTypeId = "SALES_ORDER"
-    * orderContext.orderName = orderJson.name
-    * orderContext.salesChannelEnumId = orderJson.salesChannelEnumId
-    * orderContext.orderDate = orderJson.createdAt
-    * orderContext.currencyUom = orderJson.currencyUom
-    * orderContext.presentmentCurrencyUom = orderJson.presentmentCurrencyUom
-    * orderContext.grandTotal = orderJson.grandTotal
-    * orderContext.remainingSubTotal = orderJson.remainingSubTotal
-    * orderContext.statusId = orderJson.statusId
-    * orderContext.productStoreId = orderJson.productStoreId
-    * orderContext.attributes = orderJson.attributes
-    * orderContext.adjustments = orderJson.adjustments
-4. Initialize orderContext.roles list
-    * Call findOrCreate#Person for orderJson.customer
-    * Add [partyId:findOrCreatePartyOutput.partyId, roleTypeId:"SHIP_TO_CUSTOMER"] to orderContext.roles
+        * payload (Map)
```

Comment:
I believe we are using key word payload to represent the json, The actual JSON will be native to how Moqui service and API handles hierarchical parameters.